dms/models/dms_file.py

```diff
@@ -617,6 +655,30 @@ def _inverse_content(self):
             for vals, ids in updates.items():
                 self.browse(ids).write(dict(vals))
 
+    def _create_model_attachment(self, vals):
+        if "default_directory_id" in self._context:
```

Comment:
Don't use `self._context`, but `self.env.context`. I see also that the context is something very weak, and more treated as a default value. It should be in `vals` already if such context is there.

Another thing is that this create must be @api.model_create_multi for not losing performance.